tests/attestation_plugin_cert/host/host.cpp

```diff
@@ -0,0 +1,219 @@
+    {
+        result = get_tls_cert_signed_with_ec_key(
+            enclave, &ecall_result, &cert, &cert_size);
+    }
+    else if (test_type == TEST_RSA_KEY)
+    {
+        result = get_tls_cert_signed_with_rsa_key(
+            enclave, &ecall_result, &cert, &cert_size);
+    }
+
+    if ((result != OE_OK) || (ecall_result != OE_OK))
+        oe_put_err(
+            "get_tls_cert_signed_with_%s_key() failed: result=%u",
+            test_type == TEST_RSA_KEY ? "rsa" : "ec",
+            result);
+
+    fflush(stdout);
+
+    {
+        // for testing purpose, output the whole cer in DER format
+        char filename[FILENAME_LENGTH];
+        FILE* file = NULL;
+
+        sprintf_s(
+            filename,
+            sizeof(filename),
+            "./cert_%s.der",
+            test_type == TEST_RSA_KEY ? "rsa" : "ec");
+        OE_TRACE_INFO(
+            "Host: Log quote embedded certificate to file: [%s]\n", filename);
+#ifdef _WIN32
+        fopen_s(&file, filename, "wb");
+#else
+        file = fopen(filename, "wb");
+#endif
+        fwrite(cert, 1, cert_size, file);
+        fclose(file);
+    }
+
+    // validate cert
+    OE_TRACE_INFO("Host: Verifying tls certificate\n");
+    OE_TRACE_INFO("Host: cert = %p cert_size = %d\n", cert, cert_size);
+    result = oe_verify_attestation_certificate_with_evidence(
+        cert, cert_size, enclave_claims_verifier, NULL);
+    OE_TRACE_INFO(
+        "Host: Verifying the certificate from a host ... %s\n",
+        result == OE_OK ? "Success" : "Fail");
+    fflush(stdout);
+    OE_TEST(result == OE_OK);
+
+    OE_TRACE_INFO("free cert 0xx%p\n", cert);
+    // oe_verifier_shutdown();
+    free(cert);
+}
+
+int main(int argc, const char* argv[])
+{
+#ifdef OE_LINK_SGX_DCAP_QL
+
+#ifdef _WIN32
+    /* This is a workaround for running in Visual Studio 2017 Test Explorer
+     * where the environment variables are not correctly propagated to the
+     * test. This is resolved in Visual Studio 2019 */
+    WCHAR path[_MAX_PATH];
+
+    if (!GetEnvironmentVariableW(L"SystemRoot", path, _MAX_PATH))
+    {
+        if (GetLastError() != ERROR_ENVVAR_NOT_FOUND)
+            exit(1);
+
+        UINT path_length = GetSystemWindowsDirectoryW(path, _MAX_PATH);
+        if (path_length == 0 || path_length > _MAX_PATH)
+            exit(1);
+
+        if (SetEnvironmentVariableW(L"SystemRoot", path) == 0)
+            exit(1);
+    }
+
+    if (!GetEnvironmentVariableW(L"LOCALAPPDATA", path, _MAX_PATH))
+    {
+        if (GetLastError() != ERROR_ENVVAR_NOT_FOUND)
+            exit(1);
+
+        WCHAR* local_path = NULL;
+        if (SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, NULL, &local_path) !=
+            S_OK)
+        {
+            exit(1);
+        }
+
+        BOOL success = SetEnvironmentVariableW(L"LOCALAPPDATA", local_path);
+        CoTaskMemFree(local_path);
+
+        if (!success)
+            exit(1);
+    }
+#endif
+
+    oe_result_t result;
+    oe_enclave_t* enclave = NULL;
```

Comment:
```suggestion
    oe_enclave_t* enclave = nullptr;
```
cpp files should use the C++ keyword `nullptr` instead of NULL

Reply:
Agree nullptr is better. Changed.